pkg/controller/worker/machines_test.go

```diff
@@ -640,6 +641,107 @@ var _ = Describe("Machines", func() {
+								},
+							},
+						},
+					}
+
+					// Initialize mocking functions
+					expectGetSecretCallToWork(c, awsAccessKeyID, awsSecretAccessKey)
+					expectDeleteAllCallToWork(c, &awsMachineClassList)
+
+					workerDelegate, _ := NewWorkerDelegate(common.NewClientContext(c, scheme, decoder), chartApplier, "", w, cluster)
+					chartApplier.
+						EXPECT().
+						Apply(
+							context.TODO(),
+							filepath.Join(aws.InternalChartsPath, "machineclass"),
+							namespace,
+							"machineclass",
+							kubernetes.Values(machineClasses),
+						).
+						Return(nil)
+
+					err := workerDelegate.DeployMachineClasses(context.TODO())
+					Expect(err).NotTo(HaveOccurred())
+
+					// Should have cleaned up all the old AWSMachineClasses
+					// If the actual code was called the wrong object kind,
+					// the desired object deletion wouldn't have occured.
+					Expect(len(awsMachineClassList.Items)).To(Equal(0))
+				})
+
+				It("should delete all the AWSMachineClasses as parameter is AWSMachineClass", func() {
+					// awsMachineClassList is the test object used to track the
+					// list of AWSMachineClass objects
+					testNamespace := "test-ns"
+					awsMachineClassList := machinev1alpha1.AWSMachineClassList{
+						Items: []machinev1alpha1.AWSMachineClass{
+							{
+								ObjectMeta: metav1.ObjectMeta{
+									Name: "test-aws-machine-class-0",
+								},
+							},
+							{
+								ObjectMeta: metav1.ObjectMeta{
+									Name: "test-aws-machine-class-1",
+								},
+							},
+						},
+					}
+
+					expectDeleteAllCallToWork(c, &awsMachineClassList)
+					commonClient := common.NewClientContext(c, scheme, decoder)
+
+					err := commonClient.Client().DeleteAllOf(ctx, &machinev1alpha1.AWSMachineClass{}, client.InNamespace(testNamespace))
+					Expect(err).NotTo(HaveOccurred())
+
+					// Should have cleaned up all the AWSMachineClasses
+					Expect(len(awsMachineClassList.Items)).To(Equal(0))
+				})
+
+				It("should not delete the any of the AWSMachineClasses as parameter is MachineClass", func() {
+					// awsMachineClassList is the test object used to track the
+					// list of AWSMachineClass objects
+					testNamespace := "test-ns"
+					awsMachineClassList := machinev1alpha1.AWSMachineClassList{
+						Items: []machinev1alpha1.AWSMachineClass{
+							{
+								ObjectMeta: metav1.ObjectMeta{
+									Name: "test-aws-machine-class-0",
+								},
+							},
+							{
+								ObjectMeta: metav1.ObjectMeta{
+									Name: "test-aws-machine-class-1",
+								},
+							},
+						},
+					}
+
+					expectDeleteAllCallToWork(c, &awsMachineClassList)
+					commonClient := common.NewClientContext(c, scheme, decoder)
+
+					err := commonClient.Client().DeleteAllOf(ctx, &machinev1alpha1.MachineClass{}, client.InNamespace(testNamespace))
+					Expect(err).ToNot(HaveOccurred())
+
+					// Should not clean up any of the AWSMachineClasses
+					Expect(len(awsMachineClassList.Items)).To(Equal(2))
```

Comment:
Shouldn't we call the function under test? Which function of workerDelegate we test in this test and in the above one? You can see that the other tests are calling a func of the workerDelegate - like `GenerateMachineDeployments`, `DeployMachineClasses`, etc.

Reply:
Hi @ialidzhikov ,

I got your concern with the unit tests. I will try to fix the issues and make the changes soon. Sorry was stuck with a few other things.

Reply:
I have tried to address this with more meaningful scenarios (failure cases) for function `DeployMachineClasses` here - https://github.com/gardener/gardener-extension-provider-aws/pull/212/commits/bb4f0948c8d40645c1258270c454229d115f6fb3#diff-12a1fa4ba4f4250731a53989c476c9407e832ccecae17ad3fc1b2eee100ad7c3R684.